callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsAuthenticator.java

```diff
@@ -167,6 +167,14 @@ public void doAuthentication() {
 		}
 	}
 	
+	public void doAuthenticationAfterExpire(long timeout) {
+		scheduler.schedule(new Runnable() {
+			public void run() {
+				doAuthentication();
+			}
+		}, timeout, TimeUnit.SECONDS);
+	}
```

Comment:
Yes i will be canceling the scheduled tasks, Yes there are multiple paths in which authentication happen but if the authentication is already in progress it does not do re-auth.